samples/provider-properties/build.gradle.kts

```diff
@@ -1,8 +1,10 @@
 apply<GreetingPlugin>()
 
 configure<GreetingPluginExtension> {
-    message = "Hi from Gradle"
-    outputFiles = files("$buildDir/a.txt", "$buildDir/b.txt")
+    message.set("Hi from Gradle")
+    outputFiles.from(
```

Comment:
:x: I really don't see how this is an improvement over the original.

Reply:
Ah, I may have misunderstood what this was doing before.  It looked like there were two non-final `outputFiles` fields where we really want a single instance for the extension and another for the task.  But it seems like there is some kind of magic happening under the covers.

What does this look like from Java/Groovy?  If we leave it as it was and in Java I do...

```
extension.getOutputFiles().from("somewhere")
hello.getOutputFiles().from("anotherplace")
```

Does `extension.getOutputFiles()` have both "somewhere" and "anotherplace"?

Reply:
Right, there is some magic provided by the delegated property which transforms the assignment into a `setFrom` invocation, see this test case https://github.com/gradle/kotlin-dsl/blob/cb44112374e36b41732ab390531b8bc29e8de327/provider/src/test/kotlin/org/gradle/kotlin/dsl/ConfigurableFileCollectionExtensionsTest.kt#L17

Which works thanks to https://github.com/gradle/kotlin-dsl/blob/cb44112374e36b41732ab390531b8bc29e8de327/provider/src/main/kotlin/org/gradle/kotlin/dsl/ConfigurableFileCollectionExtensions.kt#L38-L37

Thus the original task configuration code:
```kotlin
outputFiles = greeting.outputFiles
```

It's semantically equivalent to the new:
```Kotlin
outputFiles.setFrom(greeting.outputFiles)
```

It trades some verbosity in the API provider for a terser DSL.